### ISSUE TITLE
Failed to find typst-ws executable in VSCode remote

### ISSUE THREAD

**Issue by hnyls2002** [Enter-tainer/typst-preview#41]
My environment is WSL ubuntu 20.04.

I just click the button at the up-right corner in my vscode, it's `Typst Preview : preview current file`, then the extension seems to not work well.

One error message says 

```
Failed to start typst-ws(typst-ws) process: Error: spawn typst-ws ENOENT
```

Another message says

```
Failed to find typst-ws executable at /home/hnyls2002/.vscode-server/extensions/mgt19937.typst-preview-0.5.0-linux-x64/out/typst-ws,maybe we didn't ship it for your platform? Using typst-ws from PATH
```

It seems that the `typst-ws` is not found. I have no idea what it is.

Does the typst preview extension require installing typst throught rust tool chains ? I actually did it, and compiling a `*.typ` file into `*.pdf` using `typst watch` command works well on my laptop.

**Comment by Enter-tainer** [comment]
**TLDR:** You may download [typst-ws-win32-x64.exe](https://github.com/Enter-tainer/typst-preview-vscode/releases/download/v0.5.0/typst-ws-win32-x64.exe) from release and put it in your $PATH. I guess it will fix this issue as a workaround. I will try to figure out how to better support vscode-remote-* enironment when I have time.

> Failed to find typst-ws executable at /home/hnyls2002/.vscode-server/extensions/mgt19937.typst-preview-0.5.0-linux-x64/out/typst-ws

I didn't test it on vscode-remote-* environments. But I guess the problem here is that the extension(running in windows) tries to execute `typst-ws` in WSL. And that is not possible.

> Does the typst preview extension require installing typst throught rust tool chains?

This extension works by compling your typst files to images (framebuffers, in effect) and send them throught websocket. These function are  implemented in `typst-ws`. We already compile and bundle `typst-ws` in the vscode extension for all major platforms.

**Comment by Enter-tainer** [comment]
> You may download [typst-ws-win32-x64.exe](https://github.com/Enter-tainer/typst-preview-vscode/releases/download/v0.5.0/typst-ws-win32-x64.exe) from release and put it in your $PATH

typo: download and rename it to `typst-ws.exe`, then put it in PATH. 

But I'm afraid that this still won't work because the `.exe` part in the filename is missing. Maybe you can switch to native Windows/Linux before this issue is resolved.

**Comment by Myriad-Dreamin** [comment]
The `vscode-clangd` detaches the executable from its extension, and performs a early checking, which would be a possible better solution:

https://github.com/clangd/vscode-clangd/blob/d0653e60b0cce91b87a55e6321f9cb442cdf5a9d/src/install.ts#L98-L111

https://github.com/clangd/node-clangd/blob/76ab3335338d6c47ab15e1cfc8329b71baec999a/src/index.ts#L91-L106

The key is that the function checks to install a suitable executable **in its inhabiting machine**.

**Comment by atlasIT** [comment]
> > You may download [typst-ws-win32-x64.exe](https://github.com/Enter-tainer/typst-preview-vscode/releases/download/v0.5.0/typst-ws-win32-x64.exe) from release and put it in your $PATH
> 
> typo: download and rename it to `typst-ws.exe`, then put it in PATH.
> 
> But I'm afraid that this still won't work because it seems that the `.exe` part in the filename is missing. Maybe you can switch to native Windows/Linux before this issue is resolved.

I can confirm this does not work.

**Comment by ertomas** [comment]
> > You may download [typst-ws-win32-x64.exe](https://github.com/Enter-tainer/typst-preview-vscode/releases/download/v0.5.0/typst-ws-win32-x64.exe) from release and put it in your $PATH
> 
> typo: download and rename it to `typst-ws.exe`, then put it in PATH.
> 
> But I'm afraid that this still won't work because it seems that the `.exe` part in the filename is missing. Maybe you can switch to native Windows/Linux before this issue is resolved.

I can confirm this does not work.

**Comment by Enter-tainer** [comment]
We are starting to understand why this is happening thanks to the work of @Bubbleioa at #83. In short, there are two bugs.

1. We build our Linux binary using ubuntu-latest (i.e. 22.04). Earlier ubuntu or linux version will fail and see error message like `Failed to find typst-ws executable at...` because of they have old glibc. 
2. We didn't forward the websocket used by sending data from the remote extension host to the webview on your local machine. This has been resolved in #83.

If you use the latest Ubuntu, remote preview will likely to work after #83 is merged. But I would suggest always using this extension on your local machine because previewing remotely introduces unnecessary network roundtrip.
